### ISSUE TITLE
request add param 

### ISSUE THREAD

**Issue by jhoysner** [overtrue/socialite#81]
Socialite::driver('facebok')
    ->with(['hd' => 'example.com'])
    ->redirect();

the answer in the handle does not have my parameter appended?

**Comment by overtrue** [comment]
```php
>>> $config = ['facebook' => ['client_id' => '', 'client_secret' => '', 'redirect' => '']]
=> [
     "facebook" => [
       "client_id" => "",
       "client_secret" => "",
       "redirect" => "",
     ],
   ]
>>> (new Overtrue\Socialite\SocialiteManager($config))->driver('facebook')->with(['hd' => 'example.com'])->redirect()->getTargetUrl();
=> "https://www.facebook.com/v2.9/dialog/oauth?client_id=&redirect_uri=&scope=email&response_type=code&hd=example.com&state=bc96c702a5feb4cb2f94ec20e8b95924000ae424"
>>>
```

**Comment by overtrue** [comment]
@jhoysner Sorry, I can't reproduce your question.

**Comment by woei66** [comment]
I faced the same problem that when I used for facebook and the return url will be appended with extra parameters "scope=email".
As the "scope=email" is existed in the url request, our code will fail if we don't request email permission in facebook apps.
For most cases, we don't need emal of users and we won't request "email" permission in our facebook apps.
Therefore, the "email" permission is not required by default in our facebook apps.

Workround solution:
1. request email permission in facebook apps
2. figure out the where the "scope=email" is added by facebook or socialite automatically
